### DISCUSSION TITLE
Custom frame limiter

### DISCUSSION THREAD

**Discussion by archhaze24** [ppy/osu#18745] [Ideas] [answered]
I'm on Linux and I use Sway with `max_render_time` option in order to decrease the input latency in osu! on wayland. However, with the value of this option where i don't feel any input latency, there is a terrible frame drops with 2x frame limiter because compositor doesn't keep up with this frame rate. On VSync everything is normal, but I can feel the latency with it. I want to tweak in-game fps to be slightly above VSync, where I can't feel the latency and do not experience frame drops. 
Can we get a custom frame limiter option in game settings?

**Comment by peppy** [comment, marked as answer]
I don't think we will add custom, but we do have plans on changing the existing setting to give better default options. We completely agree the current settings are not great.

**Reply by huupoke** [reply]
I think adding a "1x refresh rate" option is a good enough solution for now. And this will also benefit users who use variable refresh rate (like FreeSync), where it only works if the game frame rate is BELOW monitor refresh rate.

**Reply by Feodor0090** [reply]
I want to bump request of "1x rate" option. Vsync works awful on **all** devices where i played osu!lazer ever. If your hardware can't run at 120 fps (2x limiter) all that you can do is to suffer from game's twiching without ability to lock frames to 60 to play smoothly.

**Reply by peppy** [reply]
We will definitely add a 1x limiter in the future. Please stay tuned.

**Reply by Alsen99** [reply]
I have a suggestion, how about making a slider with increments of 0.5x from 1x to 16x ( maximum of 960 update fps assuming the base frame rate is 60 update fps)

**Reply by ghost** [reply]
you shouldn't need more than 4x refresh rate

**Reply by tovc** [reply]
either we're gonna use an ingame solution, or an external one such as RTSS. being that the latter is a lot messier, I think that an ingame solution would be nicer

**Reply by its0ka** [reply]
> you shouldn't need more than 4x refresh rate

@loma-baldsson 
Definitely not true if you have 60hz display. 1000fps sometimes is not high enough (on very fast sliders or cursor move) and tearing is visible.

@peppy any news? I can't reach 2x fps limit and my tablet is overheating and throttling. Custom fps limit should be available at least in the config file. V-sync doesn't work right for some reason, there is a ton of input lag and I get only 20fps. Using libstrangle currently, but its external and probably is worse than in-game limiter

**Reply by LeaPhant** [reply]
Would be nice to have a setting slightly below your monitor's refresh rate to make use of FreeSync/G-Sync.

**Comment by huupoke** [comment]
I think you can use a 3rd party frame limiter such as: https://gitlab.com/torkel104/libstrangle

**Reply by archhaze24** [reply]
sorry for the late reply, but i tried this. unfortunately, osu immediately crashes whenever i try to use libstrangle with it.

**Comment by its0ka** [comment]
> sorry for the late reply, but i tried this. unfortunately, osu immediately crashes whenever i try to use libstrangle with it.

@archhaze24 
Who marked this off-topic? This us useful. Libstrangle works for me. I use it on appimage

**Comment by tovc** [comment]
Since trying to run osu! on a 300Hz display, I've come to realise just how inadequate the current caps are. I have the ability to run the game at 300fps, 600fps or 1000fps. Nothing in between; nothing below.

I'm currently running the game on a laptop, so the song selection doesn't run at 600/1000fps. As a result, the device runs hotter than it probably should. I'm aware that optimisations are always coming, but I'd at least appreciate an option to have a different, lower fps cap for menus

**Reply by peppy** [reply]
We'd never offer the option to run below refresh rate, which is what it sounds like you are asking for. Could you share your laptop specs? Feels very weird it can't hit 300fps but has a 300hz display.

**Reply by tovc** [reply]
running below the refresh rate during gameplay is silly, but I think it makes perfect sense during menus, at least within the context of power saving for laptops

the laptop is a 2020 HP omen laptop - i7-10750h, RTX 2060. Gameplay runs fine; 1000fps all the time, but song select is usually ~150-250fps

the display is a built-in 300hz display, not an external one. One of the main criticisms against this model is that a 2060 isn't really suitable to run much at 300hz; osu and a handful of other rhythm games being both the exception, and the reason that I bought this laptop

**Reply by peppy** [reply]
Thanks for the specs.

We'll continue to optimise, hopefully to the point song select also runs fine and doesn't suck power.
